lib/facter/puppet_settings.rb

```diff
@@ -0,0 +1,7 @@
+require 'puppet'
+
+Facter.add('puppet_confdir') do
+  setcode do
+    Puppet.settings['confdir']
```

Comment:
Personally I would prefer it to move the facts from puppet_agent to extlib/stdlib, or to extend the extlib fact with `confdir`, instead of another module having facts related to the puppet config.

Reply:
Fair enough, I'll see about getting extlib updated.